Makie/src/dim-converts/dim-converts.jl

```diff
@@ -66,17 +66,41 @@ end
 
 # Return instance of AbstractDimConversion for a given type
 create_dim_conversion(argument_eltype::DataType) = NoDimConversion()
-should_dim_convert(::Type{<:Real}) = false
+should_dim_convert() = nothing
 function convert_dim_observable(::NoDimConversion, value::Observable, deregister)
     return value
 end
 
 # get_ticks needs overloading for Dim Conversion
 # Which gets ignored for no conversion/nothing
-function get_ticks(::Union{Nothing, NoDimConversion}, ticks, scale, formatter, vmin, vmax)
+function get_ticks(::Union{Nothing, NoDimConversion}, ticks, scale, formatter, vmin, vmax, show_in_label)
     return get_ticks(ticks, scale, formatter, vmin, vmax)
 end
 
+# TODO: temporary
+function get_ticks(c::Union{Nothing, AbstractDimConversion}, ticks, scale, formatter, vmin, vmax)
+    return get_ticks(c, ticks, scale, formatter, vmin, vmax, true)
+end
+
+show_dim_convert_in_ticklabel(dc::Union{AbstractDimConversion, Nothing}, ::Automatic) = show_dim_convert_in_ticklabel(dc)
+show_dim_convert_in_ticklabel(::Union{AbstractDimConversion, Nothing}) = false
+show_dim_convert_in_ticklabel(::Union{AbstractDimConversion, Nothing}, option::Bool) = option
+
+# Should this trigger an error or just return ""?
+get_label_suffix(dc, format, use_short) = apply_format(get_label_suffix(dc, use_short), format)
+get_label_suffix(dc, format) = apply_format(get_label_suffix(dc), format)
+get_label_suffix(dc, ::Bool) = get_label_suffix(dc)
+get_label_suffix(dc) = error("No axis label suffix defined for conversion $dc.")
+get_label_suffix(dc::Union{Nothing, NoDimConversion}) = ""
```

Comment:
I set this up to error when setting `ax.x_unit_in_label = true` for a dim convert that doesn't implement `get_label_suffix`. We could also have it return an empty suffix, but then we would probably want to not automatically apply the format. Because that results in `"[]"`.
And if this does error we should probably remove the `Nothing/NoDimConversion` case.